### PR TITLE
Use max contrast variable to cap the generated colours

### PR DESCRIPTION
**Pull request by skjnldsv**
Before
Capture d’écran_2019-03-21_08-46-48

After
Capture d’écran_2019-03-21_08-45-31

@nextcloud/designers

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -1532,15 +1532,37 @@
...
 			if (typeof(fileData.size) !== 'undefined' && fileData.size >= 0) {
 				simpleSize = humanFileSize(parseInt(fileData.size, 10), true);
 				// rgb(118, 118, 118) / #767676
 				// min. color contrast for normal text on white background according to WCAG AA
-				sizeColor = Math.round(118-Math.pow((fileData.size/(1024*1024)),2));
+				sizeColor = Math.round(118-Math.pow((fileData.size/(1024*1024)), 2));
+
+				// ensure that the brightest color is still readable
+				// min. color contrast for normal text on white background according to WCAG AA
+				if (sizeColor >= maxContrast) {
```
Comment (ChristophWurst) [inline apps/files/js/filelist.js]: You can use `Math.min` and `Math.max` to your advantage

```x = Math.max(min, Math.min(max, x))```

**Review by ChristophWurst — APPROVED** [review]
Code could be improved, but makes sense :+1:

File: apps/files/js/filelist.js (hunk 1 above)
Comment (MorrisJobke) [reply apps/files/js/filelist.js]: Let's do this in a follow-up PR.

**Pull request merged**